frontend/util.lua

```diff
@@ -304,6 +304,31 @@ function util.isEmptyDir(path)
     return true
 end
 
+--- Checks if the given path exists. Doesn't care if it's a file or directory.
+---- @string path
+---- @treturn bool
+function util.pathExists(path)
+    local lfs = require("libs/libkoreader-lfs")
+    return lfs.attributes(path, "mode") ~= nil
+end
+
+--- As `mkdir -p`.
+--- Unlike lfs.mkdir(), does not error if the directory already exists, and
+--- creates intermediate directories as needed.
+---- @string path the directory to create
+---- @treturn bool true on success; nil,err_message on error
+function util.makeDirs(path)
```

Comment:
Shall we get rid of the -s? It'd make me think you can pass a table.

Reply:
I stole the name from [python's os.makedirs function](https://docs.python.org/2/library/os.html#os.makedirs), where the -s differentiates between this function (which can create intermediate directories and ignores existing ones) and `mkdir()` (which does neither).

I'm not personally invested in the `makeDirs` name specifically, but I would like something to differentiate it from lower-level "create a single directory" functions like `lfs.mkdir`.

Reply:
Imo that's exactly what makeDir (as opposed to mkdir) does. ;-)

Anyway, wouldn't some variation on make*Path*/mkpath be a much clearer name then?